src/ai/backend/manager/services/keypair_resource_policy/actions/modify_keypair_resource_policy.py

```diff
@@ -6,71 +6,51 @@
 from ai.backend.manager.services.keypair_resource_policy.actions.base import (
     KeypairResourcePolicyAction,
 )
-from ai.backend.manager.types import OptionalState, TriState
+from ai.backend.manager.types import OptionalState, PartialModifier, TriState
 
 
 @dataclass
-class ModifyKeyPairResourcePolicyInputData:
-    allowed_vfolder_hosts: OptionalState[dict[str, Any]] = field(
-        default_factory=lambda: OptionalState.nop("allowed_vfolder_hosts")
-    )
-    default_for_unspecified: OptionalState[str] = field(
-        default_factory=lambda: OptionalState.nop("default_for_unspecified")
-    )
-    idle_timeout: OptionalState[int] = field(
-        default_factory=lambda: OptionalState.nop("idle_timeout")
-    )
-    max_concurrent_sessions: OptionalState[int] = field(
-        default_factory=lambda: OptionalState.nop("max_concurrent_sessions")
-    )
-    max_containers_per_session: OptionalState[int] = field(
-        default_factory=lambda: OptionalState.nop("max_containers_per_session")
-    )
-    max_pending_session_count: OptionalState[int] = field(
-        default_factory=lambda: OptionalState.nop("max_pending_session_count")
-    )
+class KeyPairResourcePolicyModifier(PartialModifier):
+    allowed_vfolder_hosts: OptionalState[dict[str, Any]] = field(default_factory=OptionalState.nop)
+    default_for_unspecified: OptionalState[str] = field(default_factory=OptionalState.nop)
+    idle_timeout: OptionalState[int] = field(default_factory=OptionalState.nop)
+    max_concurrent_sessions: OptionalState[int] = field(default_factory=OptionalState.nop)
+    max_containers_per_session: OptionalState[int] = field(default_factory=OptionalState.nop)
+    max_pending_session_count: OptionalState[int] = field(default_factory=OptionalState.nop)
```

Comment:
This should be `TriState` since the column is nullable.

```suggestion
    max_pending_session_count: TriState[int] = field(default_factory=TriState.nop)
```